sdk/src/main/java/com/hedera/hashgraph/sdk/MirrorNodeContractQuery.java

```diff
@@ -0,0 +1,208 @@
+     *
+     * @param client
+     * @throws ExecutionException
+     * @throws InterruptedException
+     */
+    public long estimate(Client client) throws ExecutionException, InterruptedException {
+        if (this.contractEvmAddress == null) {
+            Objects.requireNonNull(this.contractId);
+            this.contractEvmAddress = getContractAddressFromMirrorNodeAsync(client, this.contractId.toString()).get();
+        }
+        return getEstimateGasFromMirrorNodeAsync(client, this.callData, this.contractEvmAddress).get();
+    }
+
+    /**
+     * Does transient simulation of read-write operations and returns the result in hexadecimal string format.
+     * The result can be any solidity type.
+     *
+     * @param client
+     * @throws ExecutionException
+     * @throws InterruptedException
+     */
+    public String call(Client client) throws ExecutionException, InterruptedException {
+        if (this.contractEvmAddress == null) {
+            Objects.requireNonNull(this.contractId);
+            this.contractEvmAddress = getContractAddressFromMirrorNodeAsync(client, this.contractId.toString()).get();
+        }
+
+        var blockNum = this.blockNumber == 0 ? "" : String.valueOf(this.blockNumber);
+        return getContractCallResultFromMirrorNodeAsync(client, this.callData, this.contractEvmAddress,
+            blockNum).get();
+    }
+
+    private static CompletableFuture<String> getContractCallResultFromMirrorNodeAsync(Client client, byte[] data,
+        String contractAddress, String blockNumber) {
+        String apiEndpoint = "/contracts/call";
+        String jsonPayload = createJsonPayload(data, contractAddress, blockNumber, false);
+        return performQueryToMirrorNodeAsync(client, apiEndpoint, jsonPayload, true)
+            .exceptionally(ex -> {
+                client.getLogger().error("Error in while performing post request to Mirror Node: " + ex.getMessage());
+                throw new CompletionException(ex);
+            })
+            .thenApply(MirrorNodeContractQuery::parseContractCallResult);
+    }
+
+    public static CompletableFuture<Long> getEstimateGasFromMirrorNodeAsync(Client client, byte[] data,
+        String contractAddress) {
+        String apiEndpoint = "/contracts/call";
+        String jsonPayload = createJsonPayload(data, contractAddress, "latest", true);
+        return performQueryToMirrorNodeAsync(client, apiEndpoint, jsonPayload, true)
+            .exceptionally(ex -> {
+                client.getLogger().error("Error in while performing post request to Mirror Node: " + ex.getMessage());
+                throw new CompletionException(ex);
+            })
+            .thenApply(MirrorNodeContractQuery::parseHexEstimateToLong);
+    }
+
+    static String createJsonPayload(byte[] data, String contractAddress, String blockNumber, boolean estimate) {
+        String hexData = Hex.toHexString(data);
+        return String.format("""
+            {
+              "data": "%s",
```

Comment:
Also missing `from`
Note `from` is also optional in the case we're simulating a contract create flow in which the data contains contract bytecode

Reply:
Added.